### PR TITLE
Fix gc crash possibly caused by system events

### PR DESCRIPTION
**Pull request by pandamicro**
Fix crash issues reported by IGS

Changes proposed in this pull request:
 * Fix gc crash possibly caused by system events

> Makes sure these boxes are checked before submitting your PR - thank you!
>
- [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- To official teams:
  - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
  - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
  - [x] Make sure any log information in `cc.log` , `cc.error`, `cc.warn` or `cc.assert` has been moved into `DebugInfos.js` with an ID

@cocos-creator/engine-admins

### CODE REVIEW

File: jsb/jsb-director.js [hunk 1]
```diff
@@ -400,29 +400,50 @@ cc.Director.EVENT_COMPONENT_UPDATE = 'director_component_update';
...
-    
-    cc.director.emit(cc.Director.EVENT_BEFORE_VISIT, this);
-});
-cc.eventManager.addCustomListener(cc.Director.EVENT_AFTER_VISIT, function () {
-    cc.director.emit(cc.Director.EVENT_AFTER_VISIT, this);
-});
-cc.eventManager.addCustomListener(cc.Director.EVENT_AFTER_DRAW, function () {
-    cc.director.emit(cc.Director.EVENT_AFTER_DRAW, this);
-});
+cc.Director._beforeUpdateListener = {
```
Comment (jareguo) [inline jsb/jsb-director.js]: 这几个变量做成局部的就行了吧

Comment (pandamicro) [reply jsb/jsb-director.js]: 不行啊，由于这几个 listener 是常驻的，用这种方式来保障它们不会被释放，以前那种局部变量的方式，在 IGS 那个比较复杂的麻将游戏中发现有崩溃的风险，虽然概率很低

Comment (jareguo) [reply jsb/jsb-director.js]: 会被释放就说明 addEventListenerWithFixedPriority 的引用出 bug 了…… 如果是为了避免这个问题就加个注释吧

Comment (pandamicro) [reply jsb/jsb-director.js]: 其实不确定是不是被释放了，无从判断，导致崩溃的情况也并不是在直接调用的地方，不过这样修改以后的确不会再崩溃了。我先加一个 comment 好了

Comment (pandamicro) [reply jsb/jsb-director.js]: 已添加

**Pull request merged**